test/teiserver_web/tachyon/matchmaking_test.exs

```diff
@@ -3,52 +3,111 @@ defmodule Teiserver.Tachyon.MatchmakingTest do
   alias Teiserver.Support.Tachyon
   alias Teiserver.OAuthFixtures
   alias Teiserver.Player
+  alias Teiserver.AssetFixtures
+  alias Teiserver.Asset
 
-  describe "list" do
-    setup {Tachyon, :setup_client}
+  def altair_attr() do
+    altair_attr("8v8")
+  end
 
-    test "works", %{client: client} do
-      resp = Tachyon.list_queues!(client)
+  def altair_attr(id),
+    do: %{
+      spring_name: "Altair Crossing Remake 1.2.3",
+      display_name: "Altair Crossing",
+      thumbnail_url: "https://www.beyondallreason.info/map/altair-crossing",
+      matchmaking_queues: ["1v1", "2v2", "3v3", id]
+    }
 
-      # convert into a set since the order must not impact test result
-      expected_playlists =
-        MapSet.new([
-          %{
-            "id" => "1v1",
-            "name" => "Duel",
-            "numOfTeams" => 2,
-            "teamSize" => 1,
-            "ranked" => true
-          },
-          %{
-            "id" => "2v2",
-            "name" => "2v2",
-            "numOfTeams" => 2,
-            "teamSize" => 2,
-            "ranked" => true
-          }
-        ])
+  def rosetta_attr(id),
+    do: %{
+      spring_name: "Rosetta",
+      display_name: "Rosetta",
+      thumbnail_url: "https://www.beyondallreason.info/map/rosetta",
+      matchmaking_queues: ["5v5", "6v6", "7v7", "8v8", id]
+    }
 
-      assert MapSet.new(resp["data"]["playlists"]) == expected_playlists
-    end
+  def map_attrs(id),
+    do: [altair_attr(id), rosetta_attr(id)]
+
+  def map_names(id) do
+    map_attrs(id)
+    |> Enum.map(fn map -> %{"springName" => map.spring_name} end)
+  end
+
+  def engine_versions(),
+    do: [%{version: "105.1.1-2590-gb9462a0 bar"}, %{version: "100.2.1-2143-test bar"}]
+
+  def game_versions(),
+    do: [
+      %{spring_game: "Beyond All Reason test-26929-d709d32"},
+      %{spring_game: "BAR test version"}
+    ]
+
+  def engine_names() do
+    engine_versions()
+    |> Enum.map(fn engine -> %{"version" => engine.version} end)
+  end
+
+  def game_names() do
+    game_versions()
+    |> Enum.map(fn game -> %{"springName" => game.spring_game} end)
   end
 
-  defp mk_queue_attrs(team_size) do
+  defp setup_queue(context) when is_map(context) do
+    setup_queue(1)
+  end
+
+  defp setup_queue(team_size) when is_integer(team_size) do
     id = UUID.uuid4()
 
+    setup_maps(id)
+    setup_queue(id, team_size)
+  end
+
+  def setup_queue(id, team_size) do
+    queue_attrs(id, team_size)
+    |> mk_queue()
+  end
+
+  def setup_queue(id, team_size, timeout) do
+    setup_maps(id)
+
+    queue_attrs(id, team_size)
+    |> put_in([:settings, :pairing_timeout], timeout)
+    |> mk_queue()
+  end
+
+  def setup_mapless_queue(id, team_size) do
+    queue_attrs(id, team_size)
+    |> Map.put(:maps, [])
+    |> mk_queue()
+  end
```

Comment:
I am not the biggest fan of these very simple function, especially since they are used only once. I think it's clearer in the tests to construct the right attrs directly, basically, inline these functions.

Reply:
Moved to tests where possible